### PR TITLE
Hana - Add username customization 

### PR DESCRIPTION
**Pull request by Zlaticanin**
Adds the ability to customize username generation for dynamic users in HanaDB.

Uses the new field username_template with the [go template language](https://golang.org/pkg/text/template/).

### CODE REVIEW

**Review by fairclothjm — COMMENTED** [review]
can we also update this doc page as well?

https://www.vaultproject.io/docs/secrets/databases#database-capabilities

**Comment by Zlaticanin** [comment]
> can we also update this doc page as well?
> 
> https://www.vaultproject.io/docs/secrets/databases#database-capabilities

Just did. Thank you!

File: website/content/docs/secrets/databases/index.mdx [hunk 1]
```diff
@@ -138,7 +138,7 @@ and private key pair to authenticate.
 | [Cassandra](/docs/secrets/databases/cassandra)        | Yes                      | Yes           | Yes (1.6+)   | Yes (1.7+)             | password                  |
 | [Couchbase](/docs/secrets/databases/couchbase)        | Yes                      | Yes           | Yes          | Yes (1.7+)             | password                  |
 | [Elasticsearch](/docs/secrets/databases/elasticdb)    | Yes                      | Yes           | Yes (1.6+)   | Yes (1.8+)             | password                  |
-| [HanaDB](/docs/secrets/databases/hanadb)              | Yes (1.6+)               | Yes           | Yes (1.6+)   | No                     | password                  |
+| [HanaDB](/docs/secrets/databases/hanadb)              | Yes (1.6+)               | Yes           | Yes (1.6+)   | Yes (1.8+)             | password                  |
```
Comment (fairclothjm) [inline website/content/docs/secrets/databases/index.mdx:141]: Should this be 1.12+ for username customization?

Comment (Zlaticanin) [reply website/content/docs/secrets/databases/index.mdx:141]: Ah yes, you are right. I will update it again right now. Thank you JM!

**Review by fairclothjm — APPROVED** [review]
LGTM

**Review by austingebauer — APPROVED** [review]
LGTM 👍

File: changelog/16631.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+```release-note:feature
```
[the remark this answers is not in the document]
Comment (Zlaticanin) [reply changelog/16631.txt:1]: Hello 👋 
Thank you for catching this. I updated the changelog: https://github.com/hashicorp/vault/pull/16986

**Pull request merged**